### PR TITLE
Removed innards and NotStrict for CompileOptions, and object Chisel

### PR DESCRIPTION
**Pull request by azidar**
Remove the deprecated `NotStrict` compile options, and the corresponding `object Chisel`.

The actual compile options trait will be removed in a followup PR.

### Contributor Checklist

- [N/A] Did you add Scaladoc to every public function/method?
- [N/A] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [N/A] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [N/A] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - code cleanup

#### API Impact

This will only impact code that relies on `import Chisel._` semantics, in that it will not compile anymore. This is in effect removal of the deprecated API.

#### Backend Code Generation Impact

None.

#### Desired Merge Strategy

 - Squash: The PR will be squashed and merged (choose this if you have no preference. 

#### Release Notes

Removal of `Chisel` and the `NotStrict` compile options.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by jackkoenig** [comment]
I want to release 5.0.0-M1 before we merge this if that's okay, it will serve as an artifact that has a Chisel API near to 3.6.0 but without SFC. I should be able to get it out by tomorrow.

**Comment by mwachs5** [comment]
im starting to try to mark things as 5.0 or 6.0. Feel free to change this to 5.0 if you want it, but if you want the M1 first we might as well put this in 6.0

**Comment by jackkoenig** [comment]
I created the v5.0.0-M1 branch but I think we should merge https://github.com/chipsalliance/chisel/pull/3064 for unique SNAPSHOTs before we merge anything major like this.

**Comment by azidar** [comment]
Changed base branch to `chisel6` to enable a later discussion for what makes it to the 5.0.0 release.

**Review by seldridge — APPROVED** [review]
Voicing my approval for landing this as part of Chisel 5. I'd like to rip this band-aid off.

**Comment by sequencer** [comment]
I personally wanna postpone it until the chisel has a commit-based release strategy, since we(RC) are having a hard time cleaning up this. I possibly want to lock a chisel version for the RC dev branch just before this commit.

**Comment by sequencer** [comment]
That will be great for us. Thank you.

**Comment by azidar** [comment]
Ok, then I'm going to merge into `main`, rather than `chisel6`.

File: core/src/main/scala/chisel3/CompileOptions.scala
Comment (azidar) [reply core/src/main/scala/chisel3/CompileOptions.scala:8]: It can, but wouldn't all references to `CompileOptions` littered everywhere in chisel3 would need to do `@nowarn`, to keep chisel warning free?

**Comment by jackkoenig** [comment]
@sequencer @azidar, We've also had per push to `main` SNAPSHOTS published since https://github.com/chipsalliance/chisel/commit/4f868dff34827f5d3cd2090d8dcf00c9a9a0e0f5, see https://s01.oss.sonatype.org/#nexus-search;gav~org.chipsalliance~chisel_2.13~~~~kw,versionexpand. Most recent is `5.0.0-M1+15-22161929-SNAPSHOT`.

I will caution that Sonatype reserves the right to clean up SNAPSHOTs, I think it generally does so in a "least used" priority order. Thus it's possible that if you depend on specific SNAPSHOT, the published artifact may not exist in the future. That being said, since the SNAPSHOT has the exact commit from which it was built, it is possible to build the correct Chisel artifact yourself in the future.

The 5.0.0-M1 tag exists but I have not yet published the artifact to Sonatype, but I am working on it right now. See https://github.com/chipsalliance/chisel/pull/3087

File: core/src/main/scala/chisel3/CompileOptions.scala
Comment (jackkoenig) [reply core/src/main/scala/chisel3/CompileOptions.scala:8]: > but wouldn't all references to CompileOptions littered everywhere in chisel3 would need to do @nowarn, to keep chisel warning free?

No, because you could add it to the `-Wconf` in scalacOptions, but it doesn't really matter since we're removing this in https://github.com/chipsalliance/chisel/pull/3056

**Comment by jackkoenig** [comment]
5.0.0-M1 has been published to Maven Central: https://central.sonatype.com/artifact/org.chipsalliance/chisel_2.13/5.0.0-M1/

Scastie: https://scastie.scala-lang.org/dKTPKwReQdS4yB2zNH2YLw
(note that you cannot generate Verilog at all with Chisel 5 because there's no SFC and firtool doesn't work in Scastie)

**Pull request merged**
